cli/command/image/build.go

```diff
@@ -162,6 +165,12 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		progBuff = bytes.NewBuffer(nil)
 		buildBuff = bytes.NewBuffer(nil)
 	}
+	if options.imageIDFile != "" {
+		// Avoid leaving a stale file if we eventually fail
+		if err := os.Remove(options.imageIDFile); err != nil && !os.IsNotExist(err) {
```

Comment:
The intention was to make `docker build --iidfile=$@` work trivially in a `Makefile` or similar without leaving stale/incorrect files around on failure and/or needing to resort to taking `|| rm -f $@` on the end of the rule.